staging/src/k8s.io/component-helpers/apimachinery/lease/controller.go

```diff
@@ -71,11 +76,14 @@ type controller struct {
 }
 
 // NewController constructs and returns a controller
-func NewController(clock clock.Clock, client clientset.Interface, holderIdentity string, leaseDurationSeconds int32, onRepeatedHeartbeatFailure func(), renewInterval time.Duration, leaseName, leaseNamespace string, newLeasePostProcessFunc ProcessLeaseFunc) Controller {
+func NewController(clock clock.Clock, client clientset.Interface, holderIdentity string, leaseDurationSeconds int32, onRepeatedHeartbeatFailure func(), renewInterval time.Duration, leaseName, leaseNamespace string, newLeasePostProcessFunc ProcessLeaseFunc, leaseRenewPolicy string) Controller {
```

Comment:
any chance this new logic can be tested with the unit tests?

Reply:
I tried, but the challenge is the backoff API https://github.com/linxiulei/kubernetes/blob/646fbe6d0a3fa1a100d90a429f039324de9c8138/staging/src/k8s.io/apimachinery/pkg/util/wait/backoff.go#L477 doesn't mock the clock API, which makes it really difficult to write unit tests

Reply:
okay, I found a way to refactor to code to workaround the difficulties. Hope it's easier to review now.